test/test_nn.py

```diff
@@ -14896,6 +14896,108 @@ def test_maxpool3d_non_square_backward(self, device):
             F.max_pool3d(x, kernel_size=(1, 1, 1)).sum().backward()
             self.assertTrue(torch.allclose(x.grad, torch.ones_like(x.grad)))
 
+    # Check that clip_grad_norm_ raises an error if the total norm of the
+    # parameters' gradients is non-finite
+    def test_clip_grad_norm_error_if_nonfinite(self, device):
```

Comment:
I rewrote this test so that it's flexible enough to test a few extra things:

- Grads with only one nonzero element, and grads with all nonzero elements
- Finite grads that overflow to infinity for higher order norms
- `parameters` input containing a second parameter with a finite grad

It might be a bit complicated, but it seemed worth it to make sure that clip_grad_norm_ does the right thing in these cases